app/routes.py

```diff
@@ -1,2 +1,328 @@
-from flask import Blueprint
+from flask import Blueprint, request, jsonify
+from werkzeug.wrappers import PlainRequest
+from app import db
+from flask.helpers import make_response
+from app.models.task import Task
+from app.models.goal import Goal
+from datetime import date
+import os
+import requests
 
+# sets up blueprints - with details
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# 
+# Wave 1: GET & POST functions - gets data for all tasks in table, and creates new tasks
+# come back and refactor into task_index & independent POST function
+@tasks_bp.route("", methods=["GET","POST"], strict_slashes=False)
+def handle_tasks():
+    if request.method == "GET":
+        task_title_from_url = request.args.get("title")
+        # search for task by title
+        if task_title_from_url:
+            tasks = Task.query.filter_by(title=task_title_from_url)
+        # all tasks
+        else:
+            tasks = Task.query.all()
+    
+        tasks_response = []
+
+        for task in tasks:
+            tasks_response.append(task.to_json())
+
+        # Wave 2 ascending/descending logic for GET method
+        if "asc" in request.full_path:
+            sorted_ascending = sorted(tasks_response, key=lambda x: x.get("title"))
+            return jsonify(sorted_ascending)
+        
+        elif "desc" in request.full_path:
+            sorted_descending = sorted(tasks_response, key=lambda x: x.get("title"), reverse=True)
+            return jsonify(sorted_descending)
+
+        return jsonify(tasks_response), 200
+
+    elif request.method == "POST":
+        # try and except block for KeyError
+        try:
+            request_body = request.get_json()
+
+            new_task = Task(title=request_body["title"],
+                            description=request_body["description"],
+                            completed_at=request_body["completed_at"]
+                            )
+
+            db.session.add(new_task)
+            db.session.commit()
+
+            return {
+                "task": new_task.to_json()
+            }, 201
+        
+        except KeyError:
+            return{"details": "Invalid data"}, 400
+
+# helper function that will eventually check for task_ids being integers
+def is_int(value):
+    try:
+        return int(value)
+    except ValueError:
+        return False
+
+# GET function - gets the data for the task with the specified task_id
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_one_task(task_id):
+    if not is_int(task_id):
+        return{
+            "message": f"ID {task_id} must be an integer",
+            "success": False
+        }, 400
+    
+    task = Task.query.get(task_id)
+
+    if task is None:
+        return make_response("", 404)
```

Comment:
We can shorten this to one line:

```suggestion
    task = Task.query.get_or_404(task_id)
```